runtime/gc_base/GCExtensions.hpp

```diff
@@ -218,8 +219,16 @@ class MM_GCExtensions : public MM_GCExtensionsBase {
 public:
 	static MM_GCExtensions* newInstance(MM_EnvironmentBase* env);
 	virtual void kill(MM_EnvironmentBase* env);
-
-	void computeDefaultMaxHeapForJava(bool enableOriginalJDK8HeapSizeCompatibilityOption);
+	/**
+	 * compute the default max heap size for java based on physical memory and other metrics
+	 * Note this function is used to adjust the max heap size during gc initialization
+	 * and to adjust the soft max value during the gc restore initialization in case the
+	 * available memory in the environment reduces
+	 * @param[in] enableOriginalJDK8HeapSizeCompatibilityOption boolean value indicates if
+	 * we enable original heap size compatibility option for JDK under JDK8
+	 * @return the computed max heap size
+	 */
```

Comment:
Please use proper syntax and punctuation (capitalize first word of the sentence, use commas, periods etc.)